Several fixes

tiagolobocastro opened the pull request:
fix(core/nexus): add children empty validation
    
    Avoids even attempting to call into the dataplane.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    fix(grpc/openapi): wrong shutdown conversion and missing shutting down
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
fix(grpc/openapi): wrong shutdown conversion and missing shutting down

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
fix(core/nexus): add children empty validation

Avoids even attempting to call into the dataplane.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

abhilashshetty04 reviewed (approved):
Lgtm but where we set it as ShuttingDown ? When we fail to shutdown during unplanned Nexus Switchover?

tiagolobocastro commented:
> Lgtm but where we set it as ShuttingDown ? When we fail to shutdown during unplanned Nexus Switchover?

Yes, it's an intermediate state reported by the dataplane.

tiagolobocastro committed:
test(snapshot/create): fix node wait fixture to include pools

Should wait until the pools are imported before marking accepting node as online.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
fix(agent/node): ignore unimplemented errors when fetching

When fetching snapshots and rebuild history, we may be talking to an older version
which does not support this, so we have to ignore unimplemented errors.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge
